### PR TITLE
Bugfix/check storage exists before attempting migration

### PR DESCRIPTION
**Pull request by patrickvacek**
Attempt to fix updater.ManualControlTests.test_manual_running_mode_once in oe-selftest.

### CODE REVIEW

File: src/libaktualizr/storage/invstorage.cc [hunk 1]
```diff
@@ -174,56 +178,97 @@ std::shared_ptr<INvStorage> INvStorage::newStorage(const StorageConfig& config,
...
 }
 
-void INvStorage::FSSToSQLS(FSStorageRead& fs_storage, SQLStorage& sql_storage) {
+bool INvStorage::FSSToSQLS(FSStorageRead& fs_storage, const std::shared_ptr<SQLStorage>& sql_storage,
+                           const bool check_exists) {
   std::string public_key;
   std::string private_key;
   if (fs_storage.loadPrimaryKeys(&public_key, &private_key)) {
-    sql_storage.storePrimaryKeys(public_key, private_key);
+    if (check_exists) {
```
Comment (patriotyk) [inline src/libaktualizr/storage/invstorage.cc]: Maybe every 
```
    if (check_exists) {
      return true;
    } else {
      sql_storage->storeTlsCa(ca);
    }
```
change to: 

```
    if (!check_exists) {
      sql_storage->storeTlsCa(ca);
    } 
```
and `return true;` at the end of method

Comment (patrickvacek) [reply src/libaktualizr/storage/invstorage.cc]: The problem is that we want to migrate if any single file from FSStorage is found, but nothing is found, we shouldn't do anything with it. Currently, the migration is attempted if `storage.path` exists even if there is nothing in it. That is wrong and it causes an error with `aktualizr-info`.

**Comment by cajun-rat** [comment]
Maybe perform the migration into a temporary, in-memory database in read-only mode?

**Comment by OYTIS** [comment]
@patrickvacek Is this PR still relevant?

**Comment by patrickvacek** [comment]
@OYTIS Yes and no. I still think something needs to be done. The only reason that oe-selftest still passes is because of the `--allow-migration` hack. The idea here is that we should only attempt a migration if there is something to migrate, not just if `/var/sota` exists. However, that is complicated by the fact that `installed_versions` is always written to devices and thus a migration _always_ happens at present. But then our tests that rely on `aktualizr-info` working before provisioning has commenced might need reworking. I haven't had time to look into this again, but I still think it's worth getting right. Ideally we would rework the `installed_versions` bit, perhaps by moving that to an `import` option in the aktualizr config. It's debatable whether most of this PR is still useful. Phil's suggestion is good, but would take a bit of work.

**Comment by patrickvacek** [comment]
Closing because of general agreement that this isn't quite the best way to go. A few of these bits might be nice, but the `installed_versions` issue stands in the way of this anyway.

**Pull request closed without merging**